rs/utils/validate_eq_derive/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::Data::Struct;
+
+/// Derive of ValidateEq trait.
+/// Derived implementation compares fields of lhs and rhs and returns the first divergence if any.
+/// Each field must have one #[validate_eq(PartialEq|Recursive|Skip)] attribute.
+///   - PartialEq compares fields of lhs and rhs using .eq() and in case of divergence
+///     reports the field name and Debug of lhs and rhs
+///   - Recursive calls .validate_eq() and ruturns its error in case of divergence
+///   - Skip ignores the field.
+
+#[derive(Debug, PartialEq)]
+enum ValidateEqFieldAttr {
+    /// Compare using .eq() and return field name if diverges.
+    PartialEq,
+    /// Call .validate_eq(); in case of deivergence return the field name and the underlying
+    /// divergence error string.
+    Recursive,
+    /// Ignore for ValidateEq
+    Skip,
+}
+
+// Find #[validate_eq(...)] attribute
+fn find_validate_eq_attr(field: &syn::Field) -> syn::Result<Option<&syn::Attribute>> {
+    let matching_attrs = field
+        .attrs
+        .iter()
+        .filter(|attr| attr.path.is_ident("validate_eq"))
+        .collect::<Vec<_>>();
+    if matching_attrs.len() == 1 {
+        Ok(Some(matching_attrs[0]))
+    } else if matching_attrs.len() == 0 {
+        Ok(None)
+    } else {
+        Err(syn::Error::new_spanned(
+            field,
+            "More than one #[validate_eq] attr",
+        ))
+    }
+}
+
+// Given the field of a named struct to derive ValidateEq for, return the attribute or panic.
```

Comment:
> or panic

It seems all branches return an error. Is this an outdated comment?

Reply:
yes